### PR TITLE
WRO-12245: Increased test coverage of VirtualList - part2

### PR DESCRIPTION
**Pull request by jeonghee27**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

Increased test coverage of VirtualList component

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Add some `VirtualList` test cases
-  I removed some native tests. (I tried to simulate scrolling for native scroll scroller. But it failed)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-12245

### Comments
Enact-DCO-1.0-Signed-off-by: Jeonghee Ahn (jeonghee27.ahn@lge.com)

### CODE REVIEW

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js [hunk 1]
```diff
@@ -228,68 +262,319 @@ describe('VirtualGridList', () => {
...
-						clientSize={clientSize}
-						dataSize={dataSize}
-						itemRenderer={renderItem}
-						itemSize={itemSize}
-						onScrollStart={handlerOnScrollStart}
-					/>
-				);
+		test('should scroll to the given align with scrollTo', (done) => {
+			const onScrollStop = handlerOnScrollStop(done, () => {
+				const expected = Math.ceil(dataSize / 7 ) * (itemSize.minHeight + 3) - clientSize.clientHeight; // 1280/180 = 7
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualGridList-specs.js:267]: Please fix all of the below.
```suggestion
				const expected = Math.ceil(dataSize / 7) * (itemSize.minHeight + 3) - clientSize.clientHeight; // 1280/180 = 7
```

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js [hunk 2]
```diff
@@ -228,68 +262,319 @@ describe('VirtualGridList', () => {
...
+		beforeEach(() => {
+			jest.useFakeTimers();
+		});
 
-				act(() => myScrollTo({position: {y: 100}, animate: false}));
+		afterEach(() => {
+			jest.useRealTimers();
+		});
+
+		test('should scroll to the by wheel', (done) => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualGridList-specs.js:464]:
```suggestion
		test('should scroll by wheel', (done) => {
```

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js [hunk 3]
```diff
@@ -228,68 +262,319 @@ describe('VirtualGridList', () => {
...
+
+			const list = screen.getByRole('list');
+			fireEvent.wheel(list, {deltaY: 100});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).toBeCalled();
+		});
+
+		test('should not scroll to the by wheel when `noScrollByWheel` prop is true', (done) => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualGridList-specs.js:493]:
```suggestion
		test('should not scroll by wheel when `noScrollByWheel` prop is true', (done) => {
```

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js [hunk 4]
```diff
@@ -228,68 +262,319 @@ describe('VirtualGridList', () => {
...
+			fireEvent.wheel(list, {deltaY: 100});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).not.toHaveBeenCalled();
+
+			done();
+		});
+
+		test('should scroll to the by drag', async () => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualGridList-specs.js:518]:
```suggestion
		test('should scroll by drag', async () => {
```

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js [hunk 5]
```diff
@@ -1,9 +1,38 @@
 import '@testing-library/jest-dom';
-import {act, render, screen} from '@testing-library/react';
+import {act, fireEvent, render, screen} from '@testing-library/react';
 
 import {VirtualGridList} from '../VirtualList';
 import {ImageItem as UiImageItem} from '../../ImageItem';
 
+const activate = (list) => fireEvent.keyUp(list, {keyCode: 13});
+const keyDown = (keyCode) => (slider) => fireEvent.keyDown(slider, {keyCode});
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualGridList-specs.js:8]:
```suggestion
const keyDown = (keyCode) => (list) => fireEvent.keyDown(list, {keyCode});
```

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js [hunk 6]
```diff
@@ -228,68 +262,319 @@ describe('VirtualGridList', () => {
...
+
+			activate(list);
+			await drag(list, {delta: {x: 0, y: -300}});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).toBeCalled();
+		});
+
+		test('should not scroll to the by key', (done) => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualGridList-specs.js:551]:
```suggestion
		test('should not scroll by key', (done) => {
```

File: packages/ui/VirtualList/tests/VirtualList-specs.js [hunk 7]
```diff
@@ -1,8 +1,37 @@
 import '@testing-library/jest-dom';
-import {act, render, screen} from '@testing-library/react';
+import {act, fireEvent, render, screen} from '@testing-library/react';
 
 import VirtualList from '../VirtualList';
 
+const activate = (list) => fireEvent.keyUp(list, {keyCode: 13});
+const keyDown = (keyCode) => (slider) => fireEvent.keyDown(slider, {keyCode});
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualList-specs.js:7]:
```suggestion
const keyDown = (keyCode) => (list) => fireEvent.keyDown(list, {keyCode});
```

File: packages/ui/VirtualList/tests/VirtualList-specs.js [hunk 8]
```diff
@@ -208,76 +244,366 @@ describe('VirtualList', () => {
...
+	describe('Scroll by event', () => {
+		beforeEach(() => {
+			jest.useFakeTimers();
+		});
+
+		afterEach(() => {
+			jest.useRealTimers();
+		});
+
+		test('should scroll to the by wheel', (done) => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualList-specs.js:493]:
```suggestion
		test('should scroll by wheel', (done) => {
```

File: packages/ui/VirtualList/tests/VirtualList-specs.js [hunk 9]
```diff
@@ -208,76 +244,366 @@ describe('VirtualList', () => {
...
+
+			const list = screen.getByRole('list');
+			fireEvent.wheel(list, {deltaY: 100});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).toBeCalled();
+		});
+
+		test('should not scroll to the by wheel when `noScrollByWheel` prop is true', (done) => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualList-specs.js:522]:
```suggestion
		test('should not scroll by wheel when `noScrollByWheel` prop is `true`', (done) => {
```

File: packages/ui/VirtualList/tests/VirtualList-specs.js [hunk 10]
```diff
@@ -208,76 +244,366 @@ describe('VirtualList', () => {
...
+			fireEvent.wheel(list, {deltaY: 100});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).not.toHaveBeenCalled();
+
+			done();
+		});
+
+		test('should scroll to the by drag', async () => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualList-specs.js:547]:
```suggestion
		test('should scroll by drag', async () => {
```

File: packages/ui/VirtualList/tests/VirtualList-specs.js [hunk 11]
```diff
@@ -208,76 +244,366 @@ describe('VirtualList', () => {
...
+
+			activate(list);
+			await drag(list, {delta: {x: 0, y: -300}});
+
+			act(() => jest.advanceTimersByTime(1000)); // Wait onScrollStop
+
+			expect(fn).toBeCalled();
+		});
+
+		test('should not scroll to the by key', (done) => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualList-specs.js:580]:
```suggestion
		test('should not scroll by key', (done) => {
```

File: packages/ui/VirtualList/tests/VirtualList-specs.js [hunk 12]
```diff
@@ -12,29 +41,34 @@ describe('VirtualList', () => {
...
 
 		getScrollTo = (scrollTo) => {
 			myScrollTo = scrollTo;
 		};
 		handlerOnScroll = () => {
 			onScrollCount++;
 		};
-		handlerOnScrollStart = () => {
+		handlerOnScrollStart = (e) => {
+			resultScrollTop = e.scrollTop;
```
Comment (MikyungKim) [inline packages/ui/VirtualList/tests/VirtualList-specs.js:71]: Is there a specific reason for this line?

Comment (jeonghee27) [reply packages/ui/VirtualList/tests/VirtualList-specs.js:71]: Oh, I intent "startScrollTop = e.scrollTop;".
It was mistake. sorry.

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js (hunk 2 above)
Comment (jeonghee27) [reply packages/ui/VirtualList/tests/VirtualGridList-specs.js:464]: Sorry. I fixed all "to the by"

File: packages/ui/VirtualList/tests/VirtualGridList-specs.js (hunk 1 above)
Comment (jeonghee27) [reply packages/ui/VirtualList/tests/VirtualGridList-specs.js]: I fixed.

**Review by MikyungKim — APPROVED** [review]
Good work! :lgtm:

**Pull request merged**